manifests/params.pp

```diff
@@ -24,7 +24,7 @@
   }
 
   # Amazon Linux already have aws cli installed:
-  if getvar('::ec2_metadata') and $::operatingsystem != 'Amazon' {
+  if getvar('::ec2_metadata') != '' and $::operatingsystem != 'Amazon' {
```

Comment:
I am skeptical of doing this under puppet 4 (or in general, but specifically under puppet 4). Does

```puppet
  if $::ec2_metadata and $::operatingsystem != 'Amazon' { 
```

work?